include/libtorrent/torrent.hpp

```diff
@@ -1179,6 +1179,8 @@ namespace libtorrent {
 		}
 		void add_suggest_piece(piece_index_t index);
 
+		client_data_t get_clientdata() const { return m_clientdata; }
```

Comment:
what about `get_client_data` vs `get_clientdata`, personal preference?

Reply:
well, not really. I suppose it should be consistent. and I think it is now. the type is called `client_data` and the variables are all called `clientdata`.
maybe `client_data` is nicer. Maybe there's a better name still. afaict, the only reason this isn't called `userdata` (which is a well established term) is because there is already an (inferior) userdata field in `add_torrent_params`, that most likely nobody uses.
An alternative name would be `cookie`, which is also well understood I think.

Reply:
> there is already an (inferior) userdata field in add_torrent_params, that most likely nobody uses.

uTorrent Web uses the existing `userdata` field to track what UI actions should be taken when an `add_torrent_alert` is received. E.g. whether a new browser tab should be launched.

Reply:
ah yeah, I guess it is passed all the way back to the `add_torrent_alert`. I was thinking that it was only used for plugins

Reply:
actually, the `client_data_t` type is *almost* API compatible with a `void*`. If it would have an explicit template conversion operator, (instead of `get()`), the only use case (I can think of) it wouldn't support is being passed along as or assigned to a `void*`. i.e. only the use case where someone would `static_cast<>` it directly to the actual type would be allowed.

Maybe that's close enough to update the existing `userdata` to use the slightly safer type.